src/test/java/org/wildfly/security/auth/client/MatchHostRuleTest.java

```diff
@@ -0,0 +1,120 @@
+        Assert.assertTrue(MatchRule.ALL.matchHost("2001:db8:85a3:0:0:8a2e::").matches(new URI("remote+http://[2001:db8:85a3:0:0:8a2e:0:0]:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("2001:db8:85a3:0:0:8a2e::").matches(new URI("remote+http://[2001:db8:85a3:0:0:8a2e::]:9990")));
+
+        // IPv4 mapped IPv6 address
+        Assert.assertTrue(MatchRule.ALL.matchHost("0:0:0:0:0:ffff:c000:0280").matches(new URI("remote+http://[0:0:0:0:0:ffff:192.0.2.128]:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("0:0:0:0:0:ffff:192.0.2.128").matches(new URI("remote+http://[::ffff:c000:0280]:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("::ffff:192.0.2.128").matches(new URI("remote+http://[0:0:0:0:0:ffff:192.0.2.128]:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("::ffff:192.0.2.128").matches(new URI("remote+http://[::ffff:192.0.2.128]:9990")));
+
+        // different case
+        Assert.assertTrue(MatchRule.ALL.matchHost("0:0:0:0:0:ffff:0:1").matches(new URI("remote+http://[0:0:0:0:0:FFFF:0:1]:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("0:0:0:0:0:FFFF:0:1").matches(new URI("remote+http://[0:0:0:0:0:ffff:0:1]:9990")));
+
+        // special addresses
+        Assert.assertTrue(MatchRule.ALL.matchHost("::1").matches(new URI("remote+http://[0:0:0:0:0:0:0:1]:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("::").matches(new URI("remote+http://[0:0:0:0:0:0:0:0]:9990")));
+    }
+
+    @Test
+    public void testIPv6PartialSpec() throws URISyntaxException {
+        // ending on colon vs digit
+        Assert.assertTrue(MatchRule.ALL.matchHost("2001:db8:85a3:0:0:8a2e:370:").matches(new URI("remote+http://[2001:db8:85a3:0:0:8a2e:370:7334]:9990")));
+        Assert.assertTrue(MatchRule.ALL.matchHost("2001:db8:85a3:0:0:8a2e:370").matches(new URI("remote+http://[2001:db8:85a3:0:0:8a2e:370:7334]:9990")));
+        Assert.assertFalse(MatchRule.ALL.matchHost("2001:db8:85a3:0:0:8a2e:37:").matches(new URI("remote+http://[2001:db8:85a3:0:0:8a2e:370:7334]:9990")));
+
+        // different case
+        Assert.assertTrue(MatchRule.ALL.matchHost("2001:DB8:85A3:0:0:8A2E:370:").matches(new URI("remote+http://[2001:db8:85a3:0:0:8a2e:370:7334]:9990")));
+
+        // shouldn't this be incorrect?
+        Assert.assertTrue(MatchRule.ALL.matchHost("2001:db8:85a3:0:0:8a2e:37").matches(new URI("remote+http://[2001:db8:85a3:0:0:8a2e:370:7334]:9990")));
+
+    }
+
+    @Test
+    public void testHostSpecIPv6Pattern() throws URISyntaxException {
+        Pattern pattern = MatchHostRule.validHostSpecIPv6Pattern;
+
+        Assert.assertTrue(pattern.matcher("2001:db8:85a3:0:0:8a2e:370:7334").matches());
+        Assert.assertTrue(pattern.matcher("2001:db8:85a3::8a2e:370:7334").matches());
+        Assert.assertTrue(pattern.matcher("2001:db8::1").matches());
```

Comment:
Textual prefix matching is not correct here.  If we do want prefix matching, we should have a separate matcher for it that takes address + netmask bits.